src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/VSSettings.cs

```diff
@@ -54,7 +55,7 @@ public VSSettings(ISolutionManager solutionManager, IMachineWideSettings machine
             SolutionManager.SolutionClosed += OnSolutionOpenedOrClosed;
         }
 
-        private void ResetSolutionSettings()
+        private bool ResetSolutionSettingsIfNeeded()
```

Comment:
Can this be called from multiple threads?  If yes, should we use `Lazy<T>` [here](https://github.com/NuGet/NuGet.Client/pull/2858/files#diff-6545c160a4f64189999990c5a6961eb6R22).

Reply:
It can be. 
I'm considering lazy for a future PR. (It's what I mentioned in standup today.)
Didn't want to cram all of the improvements